cmd/config/internal/commands/annotate.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package commands
+
+import (
+	"strings"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/cmd/config/internal/generateddocs/commands"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// NewAnnotateRunner returns a command runner.
+func NewAnnotateRunner(parent string) *AnnotateRunner {
+	r := &AnnotateRunner{}
+	c := &cobra.Command{
+		Use:     "annotate [DIR]",
+		Args:    cobra.MaximumNArgs(1),
+		Short:   commands.AnnotateShort,
+		Long:    commands.AnnotateLong,
+		Example: commands.AnnotateExamples,
+		RunE:    r.runE,
+	}
+	fixDocs(parent, c)
+	r.Command = c
+	c.Flags().StringVar(&r.Kind, "kind", "", "Resource kind to annotate")
+	c.Flags().StringVar(&r.ApiVersion, "apiVersion", "", "Resource apiVersion to annotate")
+	c.Flags().StringVar(&r.Name, "name", "", "Resource name to annotate")
+	c.Flags().StringVar(&r.Namespace, "namespace", "", "Resource namespace to annotate")
+	c.Flags().StringSliceVar(&r.Values, "kv", []string{}, "annotation as KEY=VALUE")
+	return r
+}
+
+func AnnotateCommand(parent string) *cobra.Command {
+	return NewAnnotateRunner(parent).Command
+}
+
+type AnnotateRunner struct {
+	Command    *cobra.Command
+	Values     []string
+	Kind       string
+	Name       string
+	ApiVersion string
+	Namespace  string
+	Path       string
+}
+
+func (r *AnnotateRunner) runE(c *cobra.Command, args []string) error {
+	var input []kio.Reader
+	var output []kio.Writer
+	if len(args) == 0 {
+		rw := &kio.ByteReadWriter{Reader: c.InOrStdin(), Writer: c.OutOrStdout()}
+		input = []kio.Reader{rw}
+		output = []kio.Writer{rw}
+	} else {
+		rw := &kio.LocalPackageReadWriter{PackagePath: args[0], NoDeleteFiles: true}
+		input = []kio.Reader{rw}
+		output = []kio.Writer{rw}
+	}
+	return handleError(c, kio.Pipeline{
+		Inputs:  input,
+		Filters: []kio.Filter{r},
+		Outputs: output}.Execute())
```

Comment:
Nit: Maybe move the `}` and the Execute call to a new line.

Reply:
Done